### ISSUE TITLE
webpack compile issue?

### ISSUE THREAD

**Issue by AlexPettyfer** [olefredrik/FoundationPress#1230]
I've just noticed some of my JS files aren't working properly. I've added them in the PATH.entries by editing the config.yml file.  If I upload the JS file in dropbox and link it that way as a test, it works fine. So I checked the JS file ones they were in dist/js. Somehow the following chunk is added on top of each JS files:

`/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// identity function for calling harmony imports with the correct context
/******/ 	__webpack_require__.i = function(value) { return value; };
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 50);
/******/ })
/************************************************************************/
/******/ ({

/***/ 22:
/***/ (function(module, exports, __webpack_require__) {

"use strict";`

I have no idea what that is and how to fix it. Anyone else experiencing this?

**Comment by conorbarclay** [comment]
Are you using Windows by any chance?

**Comment by AlexPettyfer** [comment]
Hi @conorbarclay yes I am. Sorry for the lack of info, I'm new at this.

**Comment by conorbarclay** [comment]
@AlexPettyfer No worries -- can you try the fix [here](https://github.com/olefredrik/FoundationPress/issues/1219#issuecomment-361344578) and report back?

**Comment by AlexPettyfer** [comment]
Thank you! - I don't see that line, I'm working on a site that uses an earlier version of FoundationPress. This is what my gulpfile.babel.js looks like: https://gist.github.com/AlexPettyfer/3aa37709f550b911c5eccabbd52a15ab

**Comment by conorbarclay** [comment]
Oh derp, I totally misunderstood your question.

The stuff you're seeing is automatically inserted by Webpack at compile time. Is the issue you're having related to your code not running properly after it has been compiled?

**Comment by AlexPettyfer** [comment]
So it's normal to see that stuff in your js files that are in in the dist/js folder? 😳 

It's just that one of my js file is not working properly after being compiled, when it's added to dist/js. 
It gives errors in the console but that's probably some debugging needed to be done on my end! Since the rest of the compiled js files are working and I thought it might be an error related to that stuff being inserted.

So after running "npm run build", should I take my files from the dist/js and copy those on the live website. Or should I run "npm run package" and take the js file from there? Thank you!

**Comment by conorbarclay** [comment]
Yep, pretty much. Either of those two methods will result in the same contents in `dist/js`. Probably easier to run `npm run build` so you don't have to unzip the package to get the files.

**Comment by AlexPettyfer** [comment]
Thank you @conorbarclay ! Sorry for this :sweat:

**Comment by conorbarclay** [comment]
No problem! :metal:

**Comment by dmcgrew** [comment]
Did you ever figure out what the problem was? I have been having so many issues with webpack screwing up my javascript.. mainly just other jQuery plugins and JS libraries. Sounds like we've been having the same trouble.

**Comment by colin-marshall** [comment]
@dmcgrew some libraries/plugins are not up-to-date and don't export as modules. For those I would just enqueue them separately.

**Comment by russelpryor** [comment]
@dmcgrew @colin-marshall same problems... webpack seems to mess up some things. I continue to get: "[0] external "jQuery" 42 bytes {0} [not cacheable]"  when running a watch.

and none of the Foundation-relevant .js seems to work.

**Comment by colin-marshall** [comment]
@russelpryor are you using external plugins/libraries?
